### PR TITLE
Remove semver lib dependency

### PR DESCRIPTION
**Pull request by pierreblanc**
The semver lib is provided by an EPEL rpm package.
We want to remove external dependency.

### CODE REVIEW

**Comment by thekad** [comment]
The PR description should describe _why_ you're making the change

**Comment by pierreblanc** [comment]
> The PR description should describe _why_ you're making the change

Sure, I just updated the description :)

**Comment by tkrishtop** [comment]
check workload preflight-green

**Comment by tkrishtop** [comment]
check workload preflight-green

**Review by tkrishtop — APPROVED** [review]
[Looks good](https://www.distributed-ci.io/jobs/c3f991a0-3148-4bfb-80f1-9ba85c0180bf/tests/4598b5e0-6d15-4b24-898a-bf9375c68d8d), thanks @pierreblanc

File: plugins/filter/ocp_compatibility.py [hunk 1]
```diff
@@ -4,61 +4,65 @@ def filters(self):
...
             '''
-            return f"{int(x.split('.')[0]) + 3}.{int(x.split('.')[1]) - 13}"
-        # parse the API after the workload installation and write down incompatible OCP versions
-        failed_versions = {k8s2ocp(api['removedInRelease']) for api in after_api}
-        version_to_removed_apis = {
-            version: {api['name'] for api in after_api if k8s2ocp(api['removedInRelease']) == version}
-            for version in failed_versions
-        }
+            k8s_major, k8s_minor = map(int, k8s_version.split('.'))
+            ocp_major, ocp_minor = k8s_major + 3, k8s_minor - 13
```
Comment (thekad) [inline plugins/filter/ocp_compatibility.py:18]: question: where do these magic numbers come from? just curious

**Review by thekad — APPROVED** [review]

File: plugins/filter/ocp_compatibility.py (hunk 1 above)
Comment (tkrishtop) [reply plugins/filter/ocp_compatibility.py:18]: How would you convert k8s version 1.25 into OCP 4.12? I know two ways:

- major + 3, minor - 13
- float(k8s_version) + 2.87, like `round(float('1.25') + 2.87, 2)` = 4.12

We should create an official Python package :-)

**Review by fredericlepied — CHANGES_REQUESTED** [review]
Be sure to bump the minor version to be able to set a dependency on a new version in the DCI agents. See doc at https://github.com/redhatci/ansible-collection-redhatci-ocp/blob/main/docs/dev.md#bumping-the-version

**Comment by tkrishtop** [comment]
> Be sure to bump the minor version to be able to set a dependency on a new version in the DCI agents. See doc at https://github.com/redhatci/ansible-collection-redhatci-ocp/blob/main/docs/dev.md#bumping-the-version

Hi @fredericlepied thank you for pointing that out. I pushed the [change](https://github.com/redhatci/ansible-collection-redhatci-ocp/compare/92a5ef9a1f6b654daf67308014a5cb1683fca75c..40b12f7cd0dc9c896dbfab25d97a141c5379d821) but not sure if that was done correctly. Before my change, the versions in galaxy.yml file and ansible-collection-redhatci-ocp.spec were different, I've set them both to 0.6.0. Could you please have a look?

**Review by fredericlepied — APPROVED** [review]

**Comment by fredericlepied** [comment]
> > Be sure to bump the minor version to be able to set a dependency on a new version in the DCI agents. See doc at https://github.com/redhatci/ansible-collection-redhatci-ocp/blob/main/docs/dev.md#bumping-the-version
> 
> Hi @fredericlepied thank you for pointing that out. I pushed the [change](https://github.com/redhatci/ansible-collection-redhatci-ocp/compare/92a5ef9a1f6b654daf67308014a5cb1683fca75c..40b12f7cd0dc9c896dbfab25d97a141c5379d821) but not sure if that was done correctly. Before my change, the versions in galaxy.yml file and ansible-collection-redhatci-ocp.spec were different, I've set them both to 0.6.0. Could you please have a look?

Looks good. Adding a check to avoid that in the future: https://github.com/redhatci/ansible-collection-redhatci-ocp/pull/252

**Comment by tkrishtop** [comment]
> > > Be sure to bump the minor version to be able to set a dependency on a new version in the DCI agents. See doc at https://github.com/redhatci/ansible-collection-redhatci-ocp/blob/main/docs/dev.md#bumping-the-version
> > 
> > 
> > Hi @fredericlepied thank you for pointing that out. I pushed the [change](https://github.com/redhatci/ansible-collection-redhatci-ocp/compare/92a5ef9a1f6b654daf67308014a5cb1683fca75c..40b12f7cd0dc9c896dbfab25d97a141c5379d821) but not sure if that was done correctly. Before my change, the versions in galaxy.yml file and ansible-collection-redhatci-ocp.spec were different, I've set them both to 0.6.0. Could you please have a look?
> 
> Looks good. Adding a check to avoid that in the future: #252

Thanks. Let's merge the version check first then.

**Review by fredericlepied — APPROVED** [review]

**Pull request merged**